base/asyncevent.jl

```diff
@@ -264,13 +264,26 @@ end
 """
     sleep(seconds)
 
-Block the current task for a specified number of seconds. The minimum sleep time is 1
-millisecond or input of `0.001`.
+Block the current task for at least specified number of seconds.
+The resolution of sleep is around 2 microsecond (2e-6 seconds), but
+as the amount of time decreases, the likelihood of the sleep exceeding the
+given amount of time increases (since the OS or julia scheduler might not
+wake up the thread/task at exactly the right time).
+
+Note that using sleep for small increments of time usually implies that you are
+doing something weird and should consider using some form of notification system.
```

Comment:
"something weird" sounds like an odd judgement call. There are very valid reasons to suspend execution for very short amounts of time, even without any expectation that other tasks run (which `sleep` also doesn't guarantee). Not all such usecases would benefit from a notification system either.

Reply:
the reason it's weird is that if you are specifying a very small number, it implies that you want to regain control after a very small amount of time. However, due to the nature of schedulers that is not actually a thing that sleep can guarentee, so you either should be using `yield` if you don't care about not coming back immediately, writing a busy loop if you do, or if you are using it in a loop to check a condition, you should be using an event system of some sort that notifies you when the thing you are waiting for completed (e.g. using FileWatching.jl if you want to get notified when a file is modified).

Reply:
Yes, I know all of that, that's why I said that there are valid usecases for such short sleeps. I'm saying that the wording is judgmental, and that's not appropriate for a docstring. All I'm asking for is to reword this a bit.